tests/parsers/test_android.py

```diff
@@ -115,246 +102,107 @@ def fixture_raw_path(root_path):
-    value2 = ['derek']*size2
-    value = np.concatenate((np.asarray(value1, dtype=object),
-                            np.asarray(value2, dtype=object)))
-    derived[key] = value
-
-    np.testing.assert_equal(derived[key, :], value)
-
-def test_android_concat(derived, pd_df):
-    """Test concat on Android data.
+    raw_path : pytest.fixture
+        Path to Android Raw measurements text log file
+    sensor_type : NavData
+        Type of NavData object
 
-    Parameters
-    ----------
-    derived : AndroidDerived2021
-        Instance of AndroidDerived2021 for testing
-    pd_df : pytest.fixture
-        pd.DataFrame for testing measurements
     """
 
-    # remove first timestamp to match
-    pd_df = pd_df[pd_df['millisSinceGpsEpoch'] != pd_df.loc[0,'millisSinceGpsEpoch']]
-
-    # extract and combine gps and glonass data
-    gps_data = derived.where("gnss_id","gps")
-    glonass_data = derived.where("gnss_id","glonass")
-    gps_glonass_navdata = gps_data.concat(glonass_data)
-    glonass_gps_navdata = glonass_data.concat(gps_data)
-
-    # combine using pandas
-    gps_df = pd_df[pd_df["constellationType"]==1]
-    glonass_df = pd_df[pd_df["constellationType"]==3]
-    gps_glonass_df = pd.concat((gps_df,glonass_df))
-    glonass_gps_df = pd.concat((glonass_df,gps_df))
-
-    for combined_navdata, combined_df in [(gps_glonass_navdata, gps_glonass_df),
-                                          (glonass_gps_navdata, glonass_gps_df)]:
-
-        # check a few rows to make sure they're equal
-        np.testing.assert_array_equal(combined_navdata["raw_pr_m"],
-                                      combined_df["rawPrM"])
-        np.testing.assert_array_equal(combined_navdata["raw_pr_sigma_m"],
-                                      combined_df["rawPrUncM"])
-        np.testing.assert_array_equal(combined_navdata["intersignal_bias_m"],
-                                      combined_df["isrbM"])
-
-def test_imu_raw(android_raw_path):
-    """Test that AndroidRawImu initialization
-
-    Parameters
-    ----------
-    android_raw_path : pytest.fixture
-        Path to Android Raw measurements text log file
-    """
-    test_imu = android.AndroidRawImu(android_raw_path)
-    isinstance(test_imu, NavData)
+    test_navdata = sensor_type(input_path = raw_path)
+    isinstance(test_navdata, NavData)
 
-    test_imu = android.AndroidRawImu(pathlib.Path(android_raw_path))
-    isinstance(test_imu, NavData)
+    test_navdata = sensor_type(pathlib.Path(raw_path))
+    isinstance(test_navdata, NavData)
 
     # raises exception if not a file path
     with pytest.raises(FileNotFoundError):
-        android.AndroidRawImu("not_a_file.txt")
+        sensor_type("not_a_file.txt")
     with pytest.raises(FileNotFoundError):
-        android.AndroidRawImu(pathlib.Path("not_a_file.txt"))
+        sensor_type(pathlib.Path("not_a_file.txt"))
 
     # raises exception if input not string or path-like
     with pytest.raises(TypeError):
```

Comment:
We probably also want to test that the returned navdata is not empty, not nan, and that the data values match the expected result.

Reply:
additional test added and bug caught!